cmd/support-scheduler/main.go

```diff
@@ -0,0 +1,96 @@
+	"github.com/edgexfoundry/edgex-go/internal/pkg/startup"
+	"net/http"
+	"os"
+	"os/signal"
+	"strconv"
+	"syscall"
+	"time"
+
+	"github.com/edgexfoundry/edgex-go"
+	"github.com/edgexfoundry/edgex-go/internal"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/usage"
+	"github.com/edgexfoundry/edgex-go/internal/support/scheduler"
+	"github.com/edgexfoundry/edgex-go/pkg/clients/logging"
+	"github.com/gorilla/context"
+)
+
+var loggingClient logger.LoggingClient
+
+func main() {
+
+	start := time.Now()
+	var useConsul bool
+	var useProfile string
+
+	flag.BoolVar(&useConsul, "consul", false, "Indicates the service should use consul.")
+	flag.BoolVar(&useConsul, "c", false, "Indicates the service should use consul.")
+	flag.StringVar(&useProfile, "profile", "", "Specify a profile other than default.")
+	flag.StringVar(&useProfile, "p", "", "Specify a profile other than default.")
+	flag.Usage = usage.HelpCallback
+	flag.Parse()
+
+	params := startup.BootParams{UseConsul: useConsul, UseProfile: useProfile, BootTimeout: internal.BootTimeoutDefault}
+	startup.Bootstrap(params, scheduler.Retry, logBeforeInit)
+
+	ok := scheduler.Init(useConsul)
+	if !ok {
+		logBeforeInit(fmt.Errorf("%s: Service bootstrap failed!", internal.SupportSchedulerServiceKey))
+		return
+	}
+
+	scheduler.LoggingClient.Info("Service dependencies resolved...")
+	scheduler.LoggingClient.Info(fmt.Sprintf("Starting %s %s ", internal.SupportSchedulerServiceKey, edgex.Version))
+
+	// Initialize the  Service Routes
+	scheduler.LoadRestRoutes()
+
+	// Bootstrap default schedulers?
+	err := scheduler.AddDefaultSchedulers()
+	if err != nil{
+		scheduler.LoggingClient.Error(fmt.Sprintf("Failed to load default schedules and events %s",err.Error()))
+	}
+
+	// Start the ticker
+	scheduler.StartTicker()
+
+	http.TimeoutHandler(nil, time.Millisecond*time.Duration(scheduler.Configuration.Service.Timeout), "Request timed out")
+	scheduler.LoggingClient.Info(scheduler.Configuration.Service.StartupMsg, "")
+
+	errs := make(chan error, 2)
+	listenForInterrupt(errs)
+	startHttpServer(errs, scheduler.Configuration.Service.Port)
+
+	// Time it took to start service
+	scheduler.LoggingClient.Info("Service started in: "+time.Since(start).String(), "")
+	scheduler.LoggingClient.Info("Listening on port: "+strconv.Itoa(scheduler.Configuration.Service.Port), "")
+	c := <-errs
+	scheduler.Destruct()
+	scheduler.LoggingClient.Warn(fmt.Sprintf("terminating: %v", c))
+}
+
+
+func logBeforeInit(err error) {
+	command.LoggingClient = logger.NewClient(internal.CoreCommandServiceKey, false, "")
+	command.LoggingClient.Error(err.Error())
+}
+
+func listenForInterrupt(errChan chan error) {
+	go func() {
+		c := make(chan os.Signal)
+		signal.Notify(c, syscall.SIGINT)
+		errChan <- fmt.Errorf("%s", <-c)
+	}()
+}
+
+func startHttpServer(errChan chan error, port int) {
+	go func() {
+		r := command.LoadRestRoutes()
+		errChan <- http.ListenAndServe(":"+strconv.Itoa(port), context.ClearHandler(r))
```

Comment:
Please note that the routes being served are coming from the command package, not the scheduler package.

Reply:
updated the line of code to use the scheduler.LoadRestRoutes()
removed the aforementioned duplication of LoadRestRoutes() as well.